src/catalog/src/information_schema/key_column_usage.rs

```diff
@@ -0,0 +1,323 @@
+            table_schema: StringVectorBuilder::with_capacity(42),
+            table_name: StringVectorBuilder::with_capacity(42),
+            column_name: StringVectorBuilder::with_capacity(42),
+            ordinal_position: UInt32VectorBuilder::with_capacity(42),
+            position_in_unique_constraint: UInt32VectorBuilder::with_capacity(42),
+            referenced_table_schema: StringVectorBuilder::with_capacity(42),
+            referenced_table_name: StringVectorBuilder::with_capacity(42),
+            referenced_column_name: StringVectorBuilder::with_capacity(42),
+        }
+    }
+
+    /// Construct the `information_schema.KEY_COLUMN_USAGE` virtual table
+    async fn make_key_column_usage(&mut self) -> Result<RecordBatch> {
+        let catalog_name = self.catalog_name.clone();
+        let catalog_manager = self
+            .catalog_manager
+            .upgrade()
+            .context(UpgradeWeakCatalogManagerRefSnafu)?;
+
+        for schema_name in catalog_manager.schema_names(&catalog_name).await? {
+            if !catalog_manager
+                .schema_exists(&catalog_name, &schema_name)
+                .await?
+            {
+                continue;
+            }
+
+            for table_name in catalog_manager
+                .table_names(&catalog_name, &schema_name)
+                .await?
+            {
+                if let Some(table) = catalog_manager
+                    .table(&catalog_name, &schema_name, &table_name)
+                    .await?
+                {
+                    let keys = &table.table_info().meta.primary_key_indices;
+                    let schema = table.schema();
+
+                    for (idx, column) in schema.column_schemas().iter().enumerate() {
+                        let constraint_name = if column.is_time_index() {
+                            "TIME INDEX"
+                        } else if keys.contains(&idx) {
+                            "PRIMARY"
+                        } else {
+                            // TODO: foreign constraint
+                            continue;
+                        };
+
+                        self.add_key_column_usage(
+                            "def",
+                            &schema_name,
+                            constraint_name,
+                            "def",
+                            &schema_name,
+                            &table_name,
+                            &column.name,
+                            idx as u32 + 1,
+                            None,
+                            None,
+                            None,
+                            None,
+                        );
+                    }
+                } else {
+                    unreachable!();
+                }
+            }
+        }
+
+        self.finish()
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    fn add_key_column_usage(
+        &mut self,
```

Comment:
The `constraint_catalog` and `table_catalog` are always `def` by default, so we don't need to pass them, and just insert the `def` value directly in this function.